src/backend/distributed/executor/multi_task_tracker_executor.c

```diff
@@ -2766,6 +2767,12 @@ JobCleanupTask(uint64 jobId)
 	jobCleanupTask->taskId = JOB_CLEANUP_TASK_ID;
 	jobCleanupTask->queryString = jobCleanupQuery->data;
 
+	/*
+	 * We currently do not take replication model into account for tasks other
+	 * than modifications. Thus, set it to invalid.
+	 */
+	jobCleanupTask->replicationModel = REPLICATION_MODEL_INVALID;
```

Comment:
Looking it again, I agree with you. I'll pick a random one and remove the remaining. Does that work?